kolibri/plugins/learn/assets/src/views/SearchChips.vue

```diff
@@ -0,0 +1,132 @@
+<template>
+
+  <div class="results-header-group">
+    <div v-for="item in items" :key="item.key + item.value" class="filter-chip">
+      <span>
+        <p class="filter-chip-text">{{ item.text }}</p>
+        <KIconButton
+          icon="close"
+          size="mini"
+          class="filter-chip-button"
+          @click="$emit('removeItem', item)"
+        />
+      </span>
+    </div>
+    <KButton
+      :text="clearAllString()"
+      appearance="basic-link"
+      class="filter-action-button"
+      @click="$emit('clearSearch')"
+    />
+  </div>
+
+</template>
+
+
+<script>
+
+  import flatMap from 'lodash/flatMap';
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import { crossComponentTranslator } from 'kolibri.utils.i18n';
+  import useChannels from '../composables/useChannels';
+  import useLanguages from '../composables/useLanguages';
+
+  export default {
+    name: 'SearchChips',
+    mixins: [commonCoreStrings],
+    setup() {
+      const { languagesMap } = useLanguages();
+      const { channelsMap } = useChannels();
+      return { channelsMap, languagesMap };
+    },
+    props: {
+      searchTerms: {
+        type: Object,
+        default: () => ({}),
+      },
+    },
+    computed: {
+      items() {
+        return flatMap(this.searchTerms, (value, key) => {
+          if (key === 'keywords') {
+            return [
+              {
+                text: value,
+                key,
+                value,
+              },
+            ];
+          }
+          return Object.keys(value).map(val => {
+            return {
+              value: val,
+              key,
+              text: this.translate(key, val),
+            };
+          });
+        });
+      },
+    },
+    created() {
+      const LibraryPageComponent = require('./LibraryPage').default;
+      this.translator = crossComponentTranslator(LibraryPageComponent);
```

Comment:
This is new!

Reply:
Yes, there were strings required to display the search chips that did not exist in the TopicsPage component, only in the LibraryPage component - but because we import the SearchChips into the LibraryPage, I didn't want to setup a circular import, so did the import here inside the closure.

Reply:
And also, I hate myself for using crossComponentTranslator.